File: C++/137. Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& n) {
        map<int,int>m;
        for(int i:n){
            m[i]++;
        }
        for(auto i=m.begin();i!=m.end();i++){
            if(i->second==1)
            return i->first;
        }
        return 0;
    }
};

File: JAVA/137. Single Number II.java
class Solution {
    public int singleNumber(int[] n) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i : n) {
            m.put(i, m.getOrDefault(i, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> i : m.entrySet()) {
            if (i.getValue() == 1) return i.getKey();
        }
        return 0;
    }
}
